### PR TITLE
SwiftLint plugin

### PR DESCRIPTION
**Pull request by mallexxx**
Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/276630244458377/1204377796839424/f
iOS PR: https://github.com/duckduckgo/iOS/pull/2227
macOS PR: https://github.com/duckduckgo/macos-browser/pull/1318
What kind of version bump will this require?: Patch

**Optional**:

Tech Design URL:
CC: @samsymons 

**Description**:
- SwiftLint added as build plugin 
- Implemented swiftlint results caching based on source files modification date

- ❗️ See https://github.com/duckduckgo/BrowserServicesKit/pull/592 for fixed linter issues

**Steps to test this PR**:
1.
1.

**OS Testing**:

* [ ] iOS
* [ ] macOS

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: .swiftlint.tests.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+disabled_rules:
+  - no_space_in_method_call
+  - multiple_closures_with_trailing_closure
+  - block_based_kvo
+  - compiler_protocol_init
+  - unused_setter_value
+  - line_length
+  - type_name
+  - force_cast
+  - force_try
```
Comment (samsymons) [inline .swiftlint.tests.yml:12]: :+1:

**Comment by samsymons** [comment]
This is really cool - @mallexxx if you have some time this week, are you able to pull changes from `main` into this branch so we can test it against the latest code?

I'm also curious, have you also tried the [official SwiftLint package plugin](https://github.com/realm/SwiftLint/blob/main/Plugins/SwiftLintPlugin/SwiftLintPlugin.swift) and did you find it insufficient?

**Comment by mallexxx** [comment]
@samsymons, I‘ve merged the upstream. Of course I‘ve checked the official plugin, in fact current implementation is based on it (as it‘s pretty straightforward) - the difference is the official plugin runs the check against all the files and my implementation caches results to improve build speed

**Comment by mallexxx** [comment]
the main problem now is to adjust the plugin to run on CI and export the linter check results

**Review by samsymons — APPROVED** [review]
The plugin worked well for me in client PRs, nice work! And being able to get SwiftLint annotations inline when working on BSK directly is great. 👏

**Pull request merged**